# Analysing Dataset 'Dataset_w6.xlsx'

# importing data

In [33]:
import pandas as pd
import numpy as np
dataw6 = pd.read_excel('Dataset_w6.xlsx')

Sample Data

In [34]:
dataw6.head(10)

country          Region  Surface area (km2)  \
0          Afghanistan    SouthernAsia              652864   
1              Albania  SouthernEurope               28748   
2              Algeria  NorthernAfrica             2381741   
3               Angola    MiddleAfrica             1246700   
4  Antigua and Barbuda       Caribbean                 442   
5            Argentina    SouthAmerica             2780400   
6              Armenia     WesternAsia               29743   
7            Australia         Oceania             7692060   
8              Austria   WesternEurope               83871   
9           Azerbaijan     WesternAsia               86600   

   Population in thousands (2017)  GDP per capita (current US$)  \
0                           35530                         623.2   
1                            2930                        3984.2   
2                           41318                        4154.1   
3                           29784                        4714.1   
4                             102                       14764.5   
5                           44271                       14564.5   
6                            2930                        3489.1   
7                           24451                       51352.2   
8                            8736                       44117.7   
9                            9828                        5438.7   

   Agricultural production index   International trade: Imports (million US$)  
0                             125                                        3568  
1                             134                                        4669  
2                             160                                       47091  
3                             175                                        8790  
4                              88                                         491  
5                             119                                       55610  
6                             135                                        3230  
7                             111                                      189406  
8                             108                                      149299  
9                             131                                        8532

The dataw6 listing various contries in the specifc region with some characteristics like area, population, GDP and Trade details.

Q1.	Find out how many countries are there in the dataset.

In [35]:
dataw6['country'].count()

196

Ans. There are 196 countries in the dataset given.

Q2.Find out how many countries with respect to the region are available in the dataset.

In [36]:
Q2 = dataw6.groupby(['Region'])['country'].count()
Q2

Region
Caribbean            16
CentralAmerica        8
CentralAsia           5
EasternAfrica        17
EasternAsia           7
EasternEurope        10
Melanesia             5
Micronesia            4
MiddleAfrica          9
NorthernAfrica        5
NorthernAmerica       4
NorthernEurope       10
Oceania               2
Polynesia             5
South-easternAsia    11
SouthAmerica         12
SouthernAfrica        5
SouthernAsia          9
SouthernEurope       12
WesternAfrica        15
WesternAsia          18
WesternEurope         7
Name: country, dtype: int64

Q3.a.which region has the highest GDP per capita (current US$)
 

In [37]:
Q3a = dataw6.groupby("Region")['GDP per capita (current US$)'].mean().round().reset_index()
Q3a

Region  GDP per capita (current US$)
0           Caribbean                       15136.0
1      CentralAmerica                        6348.0
2         CentralAsia                        4330.0
3       EasternAfrica                        2089.0
4         EasternAsia                       27968.0
5       EasternEurope                        9292.0
6           Melanesia                        9262.0
7          Micronesia                        5008.0
8        MiddleAfrica                        3929.0
9      NorthernAfrica                        3935.0
10    NorthernAmerica                       57659.0
11     NorthernEurope                       42098.0
12            Oceania                       44823.0
13          Polynesia                        8687.0
14  South-easternAsia                       10299.0
15       SouthAmerica                        8500.0
16     SouthernAfrica                        4192.0
17       SouthernAsia                        2968.0
18     SouthernEurope                       14463.0
19      WesternAfrica                        1013.0
20        WesternAsia                       17014.0
21      WesternEurope                       55387.0

In [38]:
Q3a.nlargest(1,'GDP per capita (current US$)')

Region  GDP per capita (current US$)
10  NorthernAmerica                       57659.0

Ans. NorthernAmerica tops in GDP

Q3b. which region has the highest International trade: Imports (million US$)

In [39]:
Q3b = dataw6.groupby("Region")['International trade: Imports (million US$)'].mean().round().reset_index()
Q3b

Region  International trade: Imports (million US$)
0           Caribbean                                      3057.0
1      CentralAmerica                                     57032.0
2         CentralAsia                                     12063.0
3       EasternAfrica                                      4726.0
4         EasternAsia                                    451995.0
5       EasternEurope                                     85244.0
6           Melanesia                                      2896.0
7          Micronesia                                        99.0
8        MiddleAfrica                                      4381.0
9      NorthernAfrica                                     35034.0
10    NorthernAmerica                                    663552.0
11     NorthernEurope                                    113407.0
12            Oceania                                    112914.0
13          Polynesia                                       432.0
14  South-easternAsia                                     98325.0
15       SouthAmerica                                     32886.0
16     SouthernAfrica                                     18006.0
17       SouthernAsia                                     58199.0
18     SouthernEurope                                     76667.0
19      WesternAfrica                                      4898.0
20        WesternAsia                                     46026.0
21      WesternEurope                                    407498.0

In [40]:
Q3b.nlargest(1,'International trade: Imports (million US$)')

Region  International trade: Imports (million US$)
10  NorthernAmerica                                    663552.0

Ans. NorthernAmerica tops in International trade: Imports (million US$)

Q4.Find out which region has more consistent with respect to GDP per capita (current US$)
Ans. WesternAfrica is more consistent in GDP.
Expalnation given below

In [41]:
Q4 = dataw6.groupby('Region')['GDP per capita (current US$)'].std()
Q4

Region
Caribbean            14579.371326
CentralAmerica        4180.440883
CentralAsia           4152.215490
EasternAfrica         3716.131770
EasternAsia          27485.120923
EasternEurope         5415.488739
Melanesia            13855.842686
Micronesia            4820.669694
MiddleAfrica          5274.846281
NorthernAfrica         974.706575
NorthernAmerica      25749.619445
NorthernEurope       20671.030185
Oceania               9233.329638
Polynesia             6956.250286
South-easternAsia    16311.810656
SouthAmerica          4273.707394
SouthernAfrica        2177.551120
SouthernAsia          2855.708179
SouthernEurope        9541.703216
WesternAfrica          815.095399
WesternAsia          18812.151459
WesternEurope        24770.467728
Name: GDP per capita (current US$), dtype: float64

In [42]:
Q4.nsmallest()

Region
WesternAfrica      815.095399
NorthernAfrica     974.706575
SouthernAfrica    2177.551120
SouthernAsia      2855.708179
EasternAfrica     3716.131770
Name: GDP per capita (current US$), dtype: float64

Ans. WesternAfrica is more consistent in GDP.